x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1078,7 +1098,7 @@ private int readMultiRegions(
                             subRangeToRead,
                             readerWithOffset(reader, fileRegion, Math.toIntExact(rangeToRead.start() - regionStart)),
                             writerWithOffset(writer, fileRegion, Math.toIntExact(rangeToWrite.start() - regionStart)),
-                            ioExecutor,
+                            executor,
```

Comment:
I am thinking of the following situation: we're requesting data from a node, and we receive an error that the commit is already uploaded, and we switch to get data from blob store. Unfortunately the current PR does not give us a capability to switch the executor mid-way during the `writer`'s execution. I wonder if it'd be better to have an alternative that allows us to switch executor?

May I recommend an alternative where we give the `RangeMissingHandler` the responsibility to fork to another executor if it wishes?

Reply:
I am not sure the intended usage of this change. If this is complement to #1579, I think what @kingherc suggests makes sense. Maybe there are other use cases you have in mind?

Reply:
Sorry, I should have provided more context.

The goal of this change is to be able to use a different thread pool when executions I/O operation in the cache. One case could also be to change the executor mid way as Iraklis explained.

But the proposed solution does not work, for the reason Iraklis mentionned and also because of how error handling works. I'm going to close this change and I'll come back with something better.